docs/en/developer/create-web-component/VUE.md

```diff
@@ -1,137 +1,448 @@
+#### Maven
+Use the appropriate package manager for your OS. These instructions were
+tested with maven version 3.6.
 
-## Trade offs
+#### Gradle
+Use the appropriate package manager for your OS. These instructions were
+tested with gradle 5.2.
 
-- Vue does not embed the framework in the component
-  - this requires a secondary import
-  - can lead to version conflict
+## 2. Generate the Vue application
+Replace `{component-name}` with the desired name for the component.
 
-## Prerequisites
+``` bash
+vue create {component-name} --default
+```
 
-- [Git][]
-- [Node.js][] (using [NVM][] is recommended)
-- [Vue CLI][]
+Install dependencies for legacy browser support in the newly generated app.
+``` bash
+cd {component-name}
 
-## Step-by-Step Guide
+npm install --save-dev @babel/{cli,plugin-transform-runtime,preset-env}
+```
 
-Start by using the following commands to setup a new Web Component project in Vue.
+## 3. Add gradle support for Maven WebJars
 
-```sh
-# clone the repository if you haven't already
-git clone https://github.com/uPortal-contrib/uPortal-web-components
+### Create gradle.properties file
 
-# install vue cli if you haven't already
-npm install --global @vue/cli
+In the root directory, create a **gradle.properties** file, with the
+following content:
+```
+group=org.webjars.npm
+```
 
-# navigate to the packages folder
-cd uPortal-web-components/@uportal
+### Create build.gradle
 
-# generate a new application template
-# replace {component-name} with the desired name for the component
-vue create {component-name} --default
+Copy **build.gradle** file from @uportal directory of **uPortal-web-components**
+project (or use one from [the appendix](#buildgradle-files)).
 
-# navigate into newly created component folder
-cd {component-name}
+https://github.com/uPortal-contrib/uPortal-web-components/blob/master/%40uportal/build.gradle
 
-# add additional dependencies for legacy browser support
-npm install --save-dev @babel/{cli,plugin-transform-runtime,preset-env}
+If using the one from uPortal-web-components, remove the subprojects line and
+its enclosing brackets; for example:
+
+``` diff
+- subprojects {
+
+    apply plugin: 'java'
+    apply plugin: 'maven'
+
+    def jsonFile = file("${projectDir}/package.json")
+    def parsedJson = new groovy.json.JsonSlurper().parseText(jsonFile.text)
+    project.version = parsedJson.version + '-SNAPSHOT'
+
+    jar {
+        baseName "uportal__${project.name}"
+        from '.'
+        exclude 'build'
+        exclude 'node_modules'
+        into "META-INF/resources/webjars/uportal__${project.name}/${project.version}"
+    }
+
+- }
+```
+
+It should look something like this:
+
+``` gradle
+    apply plugin: 'java'
+    apply plugin: 'maven'
+
+    def jsonFile = file("${projectDir}/package.json")
+    def parsedJson = new groovy.json.JsonSlurper().parseText(jsonFile.text)
+    project.version = parsedJson.version + '-SNAPSHOT'
+
+    jar {
+        baseName "uportal__${project.name}"
```

Comment:
A possible future enhancement would be to get the `baseName` from the _package.json_ so the instructions can handle different namespaces other than `uportal__` and non-namespaced packages.

Potentially something like

```gradle
baseName parsedJson.name.replace('@', '').replace('/', '__')
```

Reply:
Would you like to amend this now?

Reply:
Let's hold off on this for now.
We can get this updated in a future PR after it has been verified.

Reply:
OK. I will resolve this.